lib/util/http/queue.ts

```diff
@@ -0,0 +1,39 @@
+import URL from 'url';
+import PQueue from 'p-queue';
+import { getRequestLimit } from './host-rules';
+
+let hostQueues: Record<string, PQueue> = {};
```

Comment:
```suggestion
let hostQueues: Record<string, PQueue> = Object.create({});
```
So we don't inherit object prototype.

We alternatively can use a `Map<string, PQueue>`, so we can make it `const` and call `hostQueues.clear()` to cleanup